src/main/java/com/denizenscript/depenizen/bukkit/objects/jobs/JobsJobTag.java

```diff
@@ -143,93 +151,169 @@ public String toString() {
 
     @Override
     public ObjectTag getObjectAttribute(Attribute attribute) {
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
 
-        if (jobProgression != null) {
-
-            if (attribute.startsWith("xp")) {
-                attribute = attribute.fulfill(1);
-
-                // <--[tag]
-                // @attribute <JobsJobTag.xp.max>
-                // @returns ElementTag(Number)
-                // @plugin Depenizen, Jobs
-                // @description
-                // Returns the maximum experience a player can get in a specified job.
-                // -->
-                if (attribute.startsWith("max")) {
-                    return new ElementTag(jobProgression.getMaxExperience())
-                            .getObjectAttribute(attribute.fulfill(1));
-                }
-
-                // <--[tag]
-                // @attribute <JobsJobTag.xp.level>
-                // @returns ElementTag(Number)
-                // @plugin Depenizen, Jobs
-                // @description
-                // Returns the current experience level a player has in a specified job.
-                // -->
-                if (attribute.startsWith("level")) {
-                    return new ElementTag(jobProgression.getLevel())
-                            .getObjectAttribute(attribute.fulfill(1));
-                }
+    @Override
+    public void applyProperty(Mechanism mechanism) {
+        adjust(mechanism);
```

Comment:
This wasn't even actually added by you, but, er, this should definitely not support properties. This should be a `Debug.echoError` and nothing more

Reply:
Oh, it worked this way because `player` for some reason is a property ... but now that you've added actual mechanisms it's no longer really the same.
 Could just add `&& !mechanism.isProperty` to `xp` and any other non-property mechanisms I suppose

Reply:
Please let me know if this is still applicable now that the property has been removed entirely

Reply:
Still needs to be removed, yes. The player property was the only reason it was ever valid to have.